### PR TITLE
Avoid numpy and matplotlib imports to install

### LINKED ISSUE #9: Installing requires numpy and matplotlib
Because of the lineid_plot import to get the version, in the `setup.py`, numpy and matplotlib need to be installed before lineid_plot. This complicates the use of lineid_plot in automated ci testing. Would you agree to change this, either reading (manually) the version from the file, with a regex, or duplicating the version ?

Also it would be good to have wheels if you can make a new release to fix this issue. I can propose a patch if it helps.

### PR DESCRIPTION
**Pull request by saimn**
Fixes #9.

@phn - Currently I just duplicated the version number in the `setup.py`. so to release a new version, you need to modify the version string in `setup.py`, `lineid_plot.py`, and `doc/conf.py`.

Another option would be to put the version in a python file that can be executed with `execfile` but this would need to move `lineid_plot.py` to a package (`lineid_plot/__init__.py`). Or the version could be in a text file (or stay in `lineid_plot.py`) and just reading (possibly with a regex), e.g:
```
In [22]: re.search('__version__ = "(.*)"', open('lineid_plot.py').read()).groups()[0]
Out[22]: '0.5'
```
What option do you prefer?

### CODE REVIEW

**Comment by phn** [comment]
Hello Simon,

I like your suggestion regarding moving lineid_plot into a package. I am planning to automate the tests using pytest's plugin for matplotlib, and moving to a package will help there.

Instead of putting everything in __init__.py is it possible to move 
 
     lineid_plot to lineid_plot/ldp.py (or a better name if you can thing of one)

and make sure that the top level functions are made available in __init__.py.
    
    # in __init__.py
    from .ldp import lineid_plot

This is just to keep surprises to a minimum for people using the library.

So in short can we move to the dir structure:

    lineid_plot
         __init__.py
         lpd.py
        version.py # this gets "exec"ed in setup.py?

Regards,
Prasanth

**Comment by saimn** [comment]
@phn - Ok, it should be good now: I moved `lineid_plot.py` to `lineid_plot/lineid_plot.py`, and the version is now only in `lineid_plot/version.py`.

File: lineid_plot/lineid_plot.py [hunk 1]
```diff
@@ -8,8 +8,9 @@
 import numpy as np
 from matplotlib import pyplot as plt
 
-__version__ = "0.5"
-__author__ = "Prasanth Nair"
+__all__ = ['plot_line_ids', 'initial_annotate_kwargs', 'initial_plot_kwargs',
+           'unique_labels', 'get_line_flux', 'get_box_loc', 'adjust_boxes',
+           'prepare_axes']
```
Comment (saimn) [inline lineid_plot/lineid_plot.py]: I put here all the methods that were public, but you may want to remove some ?

File: setup.py [hunk 2]
```diff
@@ -1,19 +1,23 @@
 #!/usr/bin/env python
 
-from distutils.core import setup
+import os
+from setuptools import setup, find_packages
```
Comment (saimn) [inline setup.py]: I switched to setuptools since it provides `find_packages` and is now the standard way to proceed (it should be available everywhere, as many packages already require it by default).

File: lineid_plot/lineid_plot.py (hunk 1 above)
Comment (phn) [reply lineid_plot/lineid_plot.py]: The public API is just the function 'plot_line_ids'. So perhaps we can have just that function in "__all__"?

File: setup.py (hunk 2 above)
Comment (phn) [reply setup.py]: Thanks!

File: lineid_plot/lineid_plot.py (hunk 1 above)
Comment (saimn) [reply lineid_plot/lineid_plot.py]: Most of the other functions are also mentioned in the docs, that's why I put all the functions (that were importable from `lineid_plots`).

Comment (phn) [reply lineid_plot/lineid_plot.py]: Oh! I haven't read the documentation in a while! We will keep the list you have.

**Comment by phn** [comment]
Hello Simon,

I have merged your PR. I will make a release over the weekend.

Thanks for your contributions.

Regards,
Prasanth

**Comment by saimn** [comment]
Great, thanks!
